src/Raven.Server/Documents/DatabaseRaftIndexNotifications.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Server.ServerWide;
+using Sparrow.Logging;
+using Sparrow.Server;
+using Sparrow.Utils;
+using static Raven.Server.Documents.DatabasesLandlord;
+
+namespace Raven.Server.Documents;
+
+public class DatabaseRaftIndexNotifications : AbstractRaftIndexNotifications
+{
+    private readonly Queue<DatabaseNotification> _recentNotifications = new Queue<DatabaseNotification>();
```

Comment:
`_recentNotifications` seems to be relevant for both. So, let's make it generic.